Revision 71901 - (view) (download) (annotate) - [selected]
Modified Fri Jul 26 09:55:07 2024 UTC (3 weeks ago) by ascherer
File length: 15439 byte(s)
Diff to previous 65870

[PDFTEX] Prepare for compilation with gcc-14.

There's a type mismatch (probably copy-paste-error) for
'pdf_font_has_space_char', defined as '^boolean' in section 821,
in sections 1502 and 1518.

See https://tug.org/pipermail/tex-live/2024-July/050773.html for the
initial bug report.

---
 texk/web2c/pdftexdir/pdftex.ch |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- texk/web2c/pdftexdir/pdftex.ch
+++ texk/web2c/pdftexdir/pdftex.ch	2024-08-16 11:29:11.854819548 +0000
@@ -179,7 +179,7 @@ pdf_font_type:=xmalloc_array(eight_bits,
 pdf_font_attr:=xmalloc_array(str_number, font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number, font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number, font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number, font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean, font_max);
 pdf_font_stretch:=xmalloc_array(integer, font_max);
 pdf_font_shrink:=xmalloc_array(integer, font_max);
 pdf_font_step:=xmalloc_array(integer, font_max);
@@ -298,7 +298,7 @@ pdf_font_type:=xmalloc_array(eight_bits,
 pdf_font_attr:=xmalloc_array(str_number,font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number,font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number,font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number,font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean,font_max);
 pdf_font_stretch:=xmalloc_array(integer,font_max);
 pdf_font_shrink:=xmalloc_array(integer,font_max);
 pdf_font_step:=xmalloc_array(integer,font_max);
